src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs

```diff
@@ -322,6 +322,70 @@ public IEnumerable<TabInfo> GetPageList(int parentId = -1, string searchKey = ""
             return pages;
         }
 
+        private IEnumerable<TabInfo> GetPageListHelper(string searchKey = "", bool includeHidden = true, bool includeDeleted = false)
+        {
+            var portalSettings = PortalController.Instance.GetCurrentPortalSettings();
+            var adminTabId = portalSettings.AdminTabId;
+
+            var tabs = TabController.GetPortalTabs(portalSettings.PortalId, adminTabId, false, includeHidden, includeDeleted, true);
+            var pages = from t in tabs
+                        where (t.ParentId != adminTabId || t.ParentId == Null.NullInteger) &&
+                                !t.IsSystem &&
+                                    (string.IsNullOrEmpty(searchKey)
+                                        || (!string.IsNullOrEmpty(searchKey) &&
+                                                (t.TabName.IndexOf(searchKey, StringComparison.OrdinalIgnoreCase) > Null.NullInteger
+                                                    || t.LocalizedTabName.IndexOf(searchKey, StringComparison.OrdinalIgnoreCase) > Null.NullInteger)))
+                        select t;
+
+            var sorted = pages.OrderBy(x => x.ParentId > -1 ? x.ParentId : x.TabID).ThenBy(x => x.TabID);
+            return sorted;
+        }
+
+        public IEnumerable<TabInfo> GetSubPagesList(bool? deleted, string tabName, string tabTitle, string tabPath, string tabSkin, bool? visible, out int total, string searchKey = "", int pageIndex = -1, int pageSize = 10)
+        {
+            pageIndex = pageIndex <= 0 ? 0 : pageIndex;
+            pageSize = pageSize > 0 && pageSize <= 100 ? pageSize : 10;
+            var tabs = GetPageListHelper(searchKey, true, deleted ?? false);

```

Comment:
This line is the only difference between `GetSubPagesList` and `GetPageList` and this entire methods is otherwise an exact copy. This is no good. 

There should be a single method with all the logic below. Maybe accepting the `tabs` as an additional param?